api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentGroupController.java

```diff
@@ -327,6 +330,63 @@ public PaymentGroupController(PaymentGroupService paymentGroupService, PaymentGr
+        @RequestHeader(value = "return-url") String returnURL,
+        @PathVariable("payment-group-reference") String paymentGroupReference,
+        @Valid @RequestBody TelephonyCardPaymentsRequest telephonyCardPaymentsRequest) throws CheckDigitException, MethodNotSupportedException {
+
+        boolean antennaFeature = featureToggler.getBooleanValue("pci-pal-antenna-feature",false);
+        LOG.info("Feature Flag Value in CardPaymentController : {}", antennaFeature);
+        if(antennaFeature) {
+            LOG.info("Inside Telephony check!!!");
+            TelephonyProviderAuthorisationResponse telephonyProviderAuthorisationResponse = pciPalPaymentService.getPaymentProviderAutorisationTokens();
+            PaymentServiceRequest paymentServiceRequest = PaymentServiceRequest.paymentServiceRequestWith()
+                .paymentGroupReference(paymentGroupReference)
+                .paymentReference(referenceUtil.getNext("RC"))
+                .ccdCaseNumber(telephonyCardPaymentsRequest.getCcdCaseNumber())
+                .currency(telephonyCardPaymentsRequest.getCurrency().getCode())
+                .siteId(telephonyCardPaymentsRequest.getSiteId())
+                .serviceType(telephonyCardPaymentsRequest.getService().getName())
+                .amount(telephonyCardPaymentsRequest.getAmount())
+                .channel(PaymentChannels.TELEPHONY.getCode())
+                .provider(PaymentProviders.PCIPAL.getCode())
+                .build();
+            LOG.info("Access Token Value in CardPaymentController : {}", telephonyProviderAuthorisationResponse.getAccessToken());
+            LOG.info("Refresh Token Value in CardPaymentController : {}", telephonyProviderAuthorisationResponse.getRefreshToken());
+            PaymentFeeLink paymentLink = delegatingPaymentService.update(paymentServiceRequest);
+            Payment payment = getPayment(paymentLink, paymentServiceRequest.getPaymentReference());
+
+                PciPalPaymentRequest pciPalPaymentRequest = PciPalPaymentRequest.pciPalPaymentRequestWith().orderAmount(telephonyCardPaymentsRequest.getAmount().toString()).orderCurrency(telephonyCardPaymentsRequest.getCurrency().getCode())
+                    .orderReference(payment.getReference()).build();
+                telephonyProviderAuthorisationResponse = pciPalPaymentService.getTelephonyProviderLink(pciPalPaymentRequest, telephonyProviderAuthorisationResponse, telephonyCardPaymentsRequest.getService().name(),returnURL);
+                LOG.info("Next URL Value in CardPaymentController : {}", telephonyProviderAuthorisationResponse.getNextUrl());
+                TelephonyCardPaymentsResponse telephonyCardPaymentsResponse = telephonyDtoMapper.toTelephonyCardPaymentsResponse(paymentLink, payment, telephonyProviderAuthorisationResponse);
+
+            // trigger Apportion based on the launch darkly feature flag
+            boolean apportionFeature = featureToggler.getBooleanValue(apportionFeatureValue, false);
```

Comment:
Apportion feature flag in prod is named as 'apportion-feature'. are we changing the flag?

Reply:
It is just a variable name. Look at line no 74 value doesn't change. This is to remove the sonar lint issue.